### PR TITLE
Omega bugfix for card trait

### PR DESCRIPTION
**Pull request by brianries**
Bugfix to omega not checking the trait of the card being played

### CODE REVIEW

File: server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export default class OmegaPartOfTheSquad extends NonLeaderUnitCard {
             targetZoneFilter: WildcardZoneName.AnyArena,
             ongoingEffect: AbilityHelper.ongoingEffects.ignoreAllAspectPenalties({
                 cardTypeFilter: WildcardCardType.NonLeaderUnit,
-                match: (card) => this.isFirstClonePlayedByControllerThisPhase(card),
+                match: (card) => card.hasSomeTrait(Trait.Clone) && this.isFirstClonePlayedByControllerThisPhase(card),
```
Comment (VJubert) [inline server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts:30]: maybe you should add this condition into isFirstClonePlayedByControllerThisPhase?

**Review by VJubert — APPROVED** [review]

File: server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts (hunk 1 above)
Comment (AMMayberry1) [reply server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts:30]: +1

Comment (brianries) [reply server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts]: moved condition

File: server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export default class OmegaPartOfTheSquad extends NonLeaderUnitCard {
             playedCardEntry.card.hasSomeTrait(Trait.Clone) &&
             playedCardEntry.card !== card
         );
-        return clonesPlayedByThisPlayerThisPhase.length === 0;
+        return card.hasSomeTrait(Trait.Clone) && clonesPlayedByThisPlayerThisPhase.length === 0;
```
Comment (VJubert) [inline server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts:51]: I think you should test card has clone trait before iterating watcher to improve performance

Comment (brianries) [reply server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts]: Done

File: server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts [hunk 3]
```diff
@@ -43,12 +43,15 @@ export default class OmegaPartOfTheSquad extends NonLeaderUnitCard {
     }
 
     private isFirstClonePlayedByControllerThisPhase(card) {
-        const clonesPlayedByThisPlayerThisPhase = this.cardsPlayedThisPhaseWatcher.getCardsPlayed((playedCardEntry) =>
-            playedCardEntry.playedBy === card.controller &&
-            playedCardEntry.card.hasSomeTrait(Trait.Clone) &&
-            playedCardEntry.card !== card
-        );
-        return clonesPlayedByThisPlayerThisPhase.length === 0;
+        if (card.hasSomeTrait(Trait.Clone)) {
```
Comment (AMMayberry1) [inline server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts:46]: Could we streamline this a bit and reduce nesting by checking the if condition first:

```typescript
if (!card.hasSomeTrait(Trait.Clone)) {
    return false;
}

// check watcher below
```

Comment (brianries) [reply server/game/cards/02_SHD/units/OmegaPartOfTheSquad.ts]: shuffled it

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**